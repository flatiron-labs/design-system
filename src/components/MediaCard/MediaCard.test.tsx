import React from 'react'
import { Grid } from '~/index'
import { render, screen } from '~/test/utils'

import { MediaCard, MediaCardProps } from '.'

const imgSrc = 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTV6DEYCmyc8---CLd9r0GFz4VLk31QYpdl2w&usqp=CAU'

describe('Button', () => {
  const renderComponent = ({ src, children }: MediaCardProps) => {
    render(
      <Grid container spacing={3}>
        <MediaCard src={src}>{children}</MediaCard>
      </Grid>
    )
  }

  it('should render children', () => {
    renderComponent({ children: <p>foo</p>, src: imgSrc })
    screen.getByText(/foo/i)
  })
})
