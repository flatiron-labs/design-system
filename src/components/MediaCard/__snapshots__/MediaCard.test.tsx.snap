// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaCard should render a custom cta element 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root sc-gKsewC kSnaNW MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-content-xs-flex-start MuiGrid-grid-xs-10 MuiGrid-grid-sm-5 MuiGrid-grid-md-3"
    >
      <img
        alt=""
        height="193"
        src="https://placekeanu.com/318/159/g"
        width="386"
      />
      <div
        class="MuiGrid-root sc-iBPRYJ dNlfDJ MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-content-xs-space-between MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <h4
            class="sc-eCssSg gwIyGi"
            style="color: rgb(0, 239, 225);"
          >
            Canvas
          </h4>
          <p
            class="sc-jSgupP fnyTMs"
          >
            this is information about gradleaders
          </p>
        </div>
        <a
          href="https://google.com"
          title="Search Google"
        >
          Google
        </a>
      </div>
    </div>
  </div>
</body>
`;

exports[`MediaCard should render a custom image element 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root sc-gKsewC kSnaNW MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-content-xs-flex-start MuiGrid-grid-xs-10 MuiGrid-grid-sm-5 MuiGrid-grid-md-3"
    >
      <img
        alt=""
        role="presentation"
        src="https://placekeanu.com/318/159/g"
      />
      <div
        class="MuiGrid-root sc-iBPRYJ dNlfDJ MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-content-xs-space-between MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <h4
            class="sc-eCssSg gwIyGi"
            style="color: rgb(0, 239, 225);"
          >
            Canvas
          </h4>
          <p
            class="sc-jSgupP fnyTMs"
          >
            this is information about gradleaders
          </p>
        </div>
        <button
          class="sc-bdfBwQ fWwvgz"
        >
          View Gradleaders
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`MediaCard should render children 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root sc-gKsewC kSnaNW MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-content-xs-flex-start MuiGrid-grid-xs-10 MuiGrid-grid-sm-5 MuiGrid-grid-md-3"
    >
      <img
        alt=""
        height="193"
        src="https://placekeanu.com/318/159/g"
        width="386"
      />
      <div
        class="MuiGrid-root sc-iBPRYJ dNlfDJ MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-content-xs-space-between MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <h4
            class="sc-eCssSg gwIyGi"
            style="color: rgb(0, 239, 225);"
          >
            Canvas
          </h4>
          <p
            class="sc-jSgupP fnyTMs"
          >
            this is information about gradleaders
          </p>
        </div>
        <button
          class="sc-bdfBwQ fWwvgz"
        >
          View Gradleaders
        </button>
      </div>
    </div>
  </div>
</body>
`;
